pkg/ocm/sca/architectures_gather.go

```diff
@@ -0,0 +1,47 @@
+package sca
+
+import (
+	"context"
+	"runtime"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Mapping of kubernetes architecture labels to the format used by SCA API
+var kubernetesArchMapping = map[string]string{
+	"386":     "x86",
+	"amd64":   "x86_64",
+	"ppc":     "ppc",
+	"ppc64":   "ppc64",
+	"ppc64le": "ppc64le",
+	"s390":    "s390",
+	"s390x":   "s390x",
+	"ia64":    "ia64",
+	"arm64":   "aarch64",
+}
+
+func getArch(arch string) string {
+	if translation, ok := kubernetesArchMapping[arch]; ok {
+		return translation
+	}
+
+	// Default to the arch of a node where operator is running
+	return kubernetesArchMapping[runtime.GOARCH]
+}
+
+// gatherArchitectures connects to K8S API to retrieve the list of
+// nodes and create a set of the present architectures
+func (c *Controller) gatherArchitectures(ctx context.Context) (map[string]struct{}, error) {
+	nodes, err := c.coreClient.Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	architectures := make(map[string]struct{})
+	for i := range nodes.Items {
+		nodeArch := nodes.Items[i].Status.NodeInfo.Architecture
```

Comment:
There are some situations where AutoScale MachineSet from zero where the secondary architecture is not present when this code 'could' run. 

@aleskandro you might be interested in this and have some insights.

Reply:
Autoscale from zero should work and be architecture-aware with AWS and Azure IPI clusters. 

It is not supported on BM infrastructures (IPI included).. I've never looked into the IBM Machine API providers though.

I'll get a better look at this PR asap